reports/pharmacy_first_report.Rmd

```diff
@@ -133,26 +133,18 @@ df_demographics_table <- population_table %>%
     levels = c("sex", "age_band", "ethnicity", "imd", "region"),
     labels = c("Sex", "Age Band", "Ethnicity", "IMD", "Region")
   )) %>%
-  mutate(subcategory = recode(subcategory,
+  mutate(
+    subcategory = recode(subcategory,
     "female" = "Female",
-    "male" = "Male"
-  ))
-
-# Addressing an ordering problem (see PR description)
-df_demographics_table <- df_demographics_table %>%
-  mutate(subcategory = if_else(
-    category == "Ethnicity",
-    factor(subcategory, levels = ethnicity_levels, labels = ethnicity_labels),
-    factor(subcategory, levels = subcategory_levels, labels = subcategory_labels)
-  )) %>%
-  arrange(category, match(subcategory, ethnicity_labels)) %>%
-  group_by(category) %>%
-  arrange(
-    category,
-    ifelse(subcategory == "Missing", Inf, as.numeric(subcategory))
-  ) %>%
-  ungroup()
-
+    "male" = "Male",
+    "Other South Asian" = "Any other Asian background",
+    "Other Black" = "Any other Black background",
+    "Other Mixed" = "Any other mixed background",
+    "All other ethnic groups" = "Any other ethnic group",
+    "Other White" = "Any other White background")) |>
+  mutate(order = c(1:23, 28, 36, 24, 29, 35, 25, 40, 30, 34, 27, 39, 26, 37, 38, 32, 31, 33)) |> 
+  arrange(order) |>
+  select(-order)
```

Comment:
@viv3ckj instead of doing the ordering through specifying the factors, we can do something a bit simpler here and just assign each row in our dataframe a number specifying its intended rownumber in the table.